test/test_transforms.py

```diff
@@ -306,6 +306,27 @@ def _test_librosa_consistency_helper(n_fft, hop_length, power, n_mels, n_mfcc, s
         _test_librosa_consistency_helper(**kwargs2)
         _test_librosa_consistency_helper(**kwargs3)
 
+    def test_resample(self):
+        input_path = os.path.join(self.test_dirpath, 'assets', 'sinewave.wav')
+        sound, sample_rate = torchaudio.load(input_path)
+
+        upsample_rate = sample_rate * 2
+        downsample_rate = sample_rate // 2
+        invalid_resample = torchaudio.transforms.Resample(sample_rate, upsample_rate, resampling_method='foo')
+
+        self.assertRaises(ValueError, invalid_resample, sound)
+
+        upsample_resample = torchaudio.transforms.Resample(sample_rate, upsample_rate, resampling_method='kaldi')
+        up_sampled = upsample_resample(sound)
+
+        # we expect the upsampled signal to have twice as many samples
+        self.assertTrue(up_sampled.size(-1) == sound.size(-1) * 2)
+
+        downsample_resample = torchaudio.transforms.Resample(sample_rate, downsample_rate, resampling_method='kaldi')
+        down_sampled = downsample_resample(sound)
+
+        # we expect the downsampled signal to have half as many samples
+        self.assertTrue(down_sampled.size(-1) == sound.size(-1) // 2)
 
```

Comment:
So far, we are only testing the shape of the signals after the transformations. What other tests could we do?

* What would happen if we do upsample then downsample? Should we get back something close to the original signal?
* If the signal has a sample rate that is divisible by 2, and I divide the frequency by 2, would the signal simply be every other sample?

Reply:
![image](https://user-images.githubusercontent.com/10252970/61245365-4e877800-a71a-11e9-8171-5294253eae2c.png)

```
If the signal has a sample rate that is divisible by 2, and I divide the frequency by 2, would the signal simply be every other sample?
```
Doesn't look like it. Blue dots is the original sample rate and red is the downsampled one. It seems to be in between every sample. 

Okay I'll add more tests to supplement the existing kaldi accuracy and size tests 😄

Reply:
Are these properties that Kaldi's resampling have?

Reply:
oh wait you're right. it is every other sample. I couldn't see that the red dot lies exactly on top of the blue one in some cases.

also added more tests.

Reply:
Cool! Have you regenerated this plot?

Reply:
It's the same plot. you just have to look very very closely to spot it
![image](https://user-images.githubusercontent.com/10252970/61303595-c9ea3780-a7b5-11e9-9707-a7e538391ed0.png)